esphome/components/http_request/__init__.py

```diff
@@ -78,6 +82,15 @@ def validate_secure_url(config):
             cv.Optional(
                 CONF_TIMEOUT, default="5s"
             ): cv.positive_time_period_milliseconds,
+            cv.Optional(CONF_RETRY, default={}): cv.Schema(
+                {
+                    cv.Optional(CONF_COUNT, 0): cv.int_,
```

Comment:
You'd never want to allow a negative value here
```suggestion
                    cv.Optional(CONF_COUNT, 0): cv.positive_int,
```